advanced_alchemy/repository/_async.py

```diff
@@ -547,7 +547,7 @@ async def get_or_upsert(
             )
         if upsert:
             for field_name, new_field_value in kwargs.items():
-                field = getattr(existing, field_name, None)
+                field = getattr(existing, field_name, MISSING)
                 if field and field != new_field_value:
```

Comment:
```suggestion
                field = getattr(existing, field_name, MISSING)
                if field is not MISSING and field != new_field_value:
```

Otherwise the check will never be `False`